src/imap.cpp

```diff
@@ -965,6 +965,30 @@ void imap::auth_login(const string& username, const string& password)
 }
 
 
+void imaps::auth_login_with_authenticate(const std::string& mechanism, const std::string& token) {
+    auto user_esc = to_astring(mechanism);
+    auto pass_esc = to_astring(token);
+    auto cmd = format("AUTHENTICATE" + TOKEN_SEPARATOR_STR + user_esc + TOKEN_SEPARATOR_STR + pass_esc);
+    dlg_->send(cmd);
+
+    bool has_more = true;
+    while (has_more)
+    {
+        string line = dlg_->receive();
+        tag_result_response_t parsed_line = parse_tag_result(line);
```

Comment:
The error response is in the JSON format:
```
{
  "status":"401",
  "schemes":"bearer",
  "scope":"https://mail.google.com/"
}
```
Does it make sense also to store the status in the Mailio exception?